go/oasis-node/cmd/debug/txsource/workload/commission.go

```diff
@@ -0,0 +1,251 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasislabs/oasis-core/go/common/crypto/signature/signers/memory"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	consensus "github.com/oasislabs/oasis-core/go/consensus/api"
+	"github.com/oasislabs/oasis-core/go/consensus/api/transaction"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+	runtimeClient "github.com/oasislabs/oasis-core/go/runtime/client/api"
+	staking "github.com/oasislabs/oasis-core/go/staking/api"
+)
+
+// NameCommission is the name of the commission schedule amendements
+// workload.
+const NameCommission = "commission"
+
+type commission struct {
+	logger *logging.Logger
+
+	rules          staking.CommissionScheduleRules
+	account        signature.Signer
+	reckonedNonce  uint64
+	fundingAccount signature.Signer
+}
+
+func (c *commission) doAmendCommissionSchedule(ctx context.Context, rng *rand.Rand, cnsc consensus.ClientBackend, stakingClient staking.Backend) error {
+	c.logger.Debug("amend commission schedule")
+
+	// Get current epoch.
+	currentEpoch, err := cnsc.GetEpoch(ctx, consensus.HeightLatest)
+	if err != nil {
+		return fmt.Errorf("GetEpoch: %w", err)
+	}
+
+	var account *staking.Account
+	account, err = stakingClient.AccountInfo(ctx, &staking.OwnerQuery{
+		Height: consensus.HeightLatest,
+		Owner:  c.account.Public(),
+	})
+	if err != nil {
+		return fmt.Errorf("stakingClient.AccountInfo %s: %w", c.account.Public(), err)
+	}
+	commissionSchedule := account.Escrow.CommissionSchedule
+
+	// Must not alter bounds before `currentEpoch + RateBoundLead + 1`
+	// interval (+2 since the epoch could have changed before this transacrion
+	// is actually submitted).
+	nextAlignedEpoch := currentEpoch + c.rules.RateBoundLead + 2
+	// Find first next epoch aligned with rateChangeInterval.
+	for ; nextAlignedEpoch%c.rules.RateChangeInterval != 0; nextAlignedEpoch++ {
+	}
+
+	maxBoundSteps := c.rules.MaxBoundSteps
+	maxRateSteps := c.rules.MaxRateSteps
+	// Check existing bounds/steps. In case there are existing steps with
+	// epoch < nextAlignedEpoch, those cannot get amended, therefore we need to
+	// count those to not go over maxAllowed limits.
+	for _, step := range commissionSchedule.Bounds {
+		if step.Start < nextAlignedEpoch {
+			maxBoundSteps--
+			continue
+		}
+		break
+	}
+	for _, step := range commissionSchedule.Rates {
+		if step.Start < nextAlignedEpoch {
+			maxRateSteps--
+			continue
+		}
+		break
+	}
+
+	// Select bound step epochs.
+	var boundStepEpochs []epochtime.EpochTime
+	nBoundSteps := rng.Intn(maxBoundSteps-1) + 1
```

Comment:
What would be "to low"? (also this is probsbly prevented since this will always be `genesis.MaxBoundSteps - "not-yet-pruned steps that cannot be amended (less than nextAlignedEpoch in future)"`

Reply:
too low is 1 or 0. I'll see what the new generator is like